sim-lib/src/lnd.rs

```diff
@@ -59,13 +60,20 @@ impl LndNode {
             .map_err(|err| LightningError::GetInfoError(err.to_string()))?
             .into_inner();
 
+        if chains.len() != 1 {
+            return Err(LightningError::GetInfoError(
+                "LND node is not connected to a single chain".to_string(),
+            ));
+        }
+
         Ok(Self {
             client,
             info: NodeInfo {
                 pubkey: PublicKey::from_str(&identity_pubkey)
                     .map_err(|err| LightningError::GetInfoError(err.to_string()))?,
                 features: parse_node_features(features.keys().cloned().collect()),
                 alias,
+                network: chains[0].network.clone(),
```

Comment:
nit: we can use `chains.swap_remove(0).network` here to avoid unnecessarily copying a value.